src/main/java/net/shadew/util/misc/ColorUtil.java

```diff
@@ -1370,21 +1370,23 @@ public static float huef(int color) {
         float cmin = MathUtil.min(r, MathUtil.min(g, b));
 
         float delta = cmax - cmin;
-        if (delta == 0) return 0;
+        float hue = 0F;
 
-        if (cmax == r) {
-            return (g - b) / delta % 6 * 60;
-        }
+        if (delta == 0) return hue;
 
-        if (cmax == g) {
-            return ((b - r) / delta + 2) * 60;
+        if (cmax == r) {
+            hue = (g - b) / delta % 6 * 60;
```

Comment:
Actually, this is the only line where hue returns negative since `(g-b)/delta` gives a value between -1 and 1; and any value below 0 here should become 6 + that value (java's modulo operator is just fucked up). Do the negativity check here instead of in the end so you don't need the `hue` variable and so you can avoid unnecessary branching.

```
if (cmax == r && g > b) {
    return (g - b) / delta * 60;
}  else if (cmax == g) {
    return ((b - r) / delta + 2) * 60;
} else if (cmax == b) {
    return ((r - g) / delta + 4) * 60;
} else {
    return ((g - b) / delta + 6) * 60;
}
```